### PR TITLE
fix: typing issues and add modular structure for job and file event definition.

### PR DESCRIPTION
**Pull request by MatehElismar**
### Description

This PR addresses two key improvements:

1. **Typescript Typing Fixes**: Corrected the TypeScript typings to accurately represent the functionality. Previously, the `onJobEvent` callback was documented to return the entire event envelope, but it actually returns only the event data.

2. **Event Schema Addition**: Integrated the schema of all existing event types into the library. Looking ahead, we plan to enhance flexibility by allowing the ability to listen for jobs not currently listed in the library.

### Changes
- `refactor`: Introduced a modular structure for job and file event definitions.
- `fix`: Updated the onJobEvent method in AlfredRealTimeClient to register callbacks that handle only the JobEvent["event"] data rather than the full JobEvent object. This streamlines the callback, focusing on the relevant event data.

### CODE REVIEW

**Comment by rroa** [comment]
This is a breaking change, and limiting since we loose access to the envelop data.

Why are we adding the change other than "convenience"?

**Comment by MatehElismar** [comment]
> This is a breaking change, and limiting since we loose access to the envelop data.
> 
> Why are we adding the change other than "convenience"?

@rroa 
This is not a breaking change. The update simply corrects the TypeScript typings to reflect the actual behavior of the code, ensuring the library can be correctly used in TypeScript projects. The underlying functionality remains the same.

I updated the PR description to explain better the change.

**Review by rroa — DISMISSED** [review]

File: src/realtime/types/index.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
+export * from "./events/base";
```
Comment (CheranthonyRD) [inline src/realtime/types/index.ts:1]: I understand that it is not necessary to export `Base.ts`.

**Review by CheranthonyRD — COMMENTED** [review]
I understand that it is not necessary to export `Base.ts`.

**Review by CheranthonyRD — APPROVED** [review]
LGTM!

**Pull request merged**